### PR TITLE
fix(site): fix group badge visual

### LINKED ISSUE #13069: Users page > groups bubbles have insufficient padding
See:

Screenshot 2024-04-24 at 2 43 55 PM

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Just used a `button` to simplify styling. 

Closes https://github.com/coder/coder/issues/13069

### CODE REVIEW

File: site/src/pages/UsersPage/UsersTable/UserGroupsCell.tsx [hunk 1]
```diff
@@ -31,18 +30,12 @@ export function UserGroupsCell({ userGroups }: GroupsCellProps) {
...
           <PopoverTrigger>
-            <Button
+            <button
               css={{
-                justifyContent: "flex-start",
-                fontSize: theme.typography.body2.fontSize,
-                lineHeight: theme.typography.body2.lineHeight,
-                fontWeight: 400,
+                cursor: "pointer",
+                backgroundColor: "transparent",
```
Comment (Parkreiner) [inline site/src/pages/UsersPage/UsersTable/UserGroupsCell.tsx:36]: Looks like this needs an explicit color now, too – the stories are showing that the text is now the same color as the background

https://www.chromatic.com/component?appId=624de63c6aacee003aa84340&csfId=pages-userspage-userstable&specName=Example&buildNumber=15193&k=66424a1ac22385f4368607db-1200px-interactive-true&h=7&b=-1

**Review by buenos-nachos — COMMENTED** [review]
I'm not fully clear on what the original issue has highlighting – I'm guessing that the interactive area for the Groups content was really small, and wasn't filling out the entire table cell. Is that basically right?

Just want to make sure before I give the approval

**Comment by BrunoQuaresma** [comment]
Basically in the light mode, there is a box shadow around the button that is strange.

File: site/src/pages/UsersPage/UsersTable/UserGroupsCell.tsx (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/pages/UsersPage/UsersTable/UserGroupsCell.tsx:36]: Good catch!

**Review by buenos-nachos — APPROVED** [review]
Changes make sense to me!

**Pull request merged**